src/dotnet/src/Mil.Navy.Nrl.Norm/NormNode.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Mil.Navy.Nrl.Norm.Buffers;
+using System.Net;
+using System.Runtime.InteropServices;
+
+namespace Mil.Navy.Nrl.Norm
+{
+    /// <summary>
+    /// A participant in a NORM protocol session
+    /// </summary>
+    public class NormNode
+    {
+        /// <summary>
+        /// When creating a session, allows the NORM implementation to attempt to pick an identifier based on the host computer's "default" IP address.
+        /// </summary>
+        public const long NORM_NODE_ANY = 0xffffffff;
+        /// <summary>
+        /// The special value NORM_NODE_NONE corresponds to an invalid (or null) node.
+        /// </summary>
+        public const int NORM_NODE_NONE = 0;
+        /// <summary>
+        /// The special value NORM_NODE_INVALID corresponds to an invalid reference.
+        /// </summary>
+        public const int NORM_NODE_INVALID = 0;
+        /// <summary>
+        /// The handle is associated to the NORM protocol engine instance.
+        /// </summary>
+        private long _handle;
+
+        /// <summary>
+        /// Parameterized contructor.
+        /// </summary>
+        /// <param name="handle">The handle is associated to the NORM protocol engine instance.</param>
+        internal NormNode(long handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// The NormNodeId identifier for the remote participant.
+        /// </summary>
+        public long Id => NormNodeGetId(_handle);
+
+        /// <summary>
+        /// The current network source address detected for packets received from remote NORM sender.
+        /// </summary>
+        public IPEndPoint Address
+        {
+            get
+            {
+                var bufferLength = 256;
+                using var buffer = ByteBuffer.AllocateDirect(bufferLength);
+                int port;
+
+                unsafe
+                {
+                    byte* addrBuffer = null;
+                    buffer.AcquirePointer(ref addrBuffer);
+                    if (!NormNodeGetAddress(_handle, (nint)addrBuffer, ref bufferLength, out port))
+                    {
+                        throw new IOException("Failed to get node address");
+                    }
+                }
+
+                var addressBytes = new byte[bufferLength];
+                buffer.ReadArray(0, addressBytes, 0, bufferLength);
+                var ipAddressText = string.Join('.', addressBytes);
+                var ipAddress = IPAddress.Parse(ipAddressText);
+
+                return new IPEndPoint(ipAddress, port);
```

Comment:
Recommend using `stackalloc` since address has a small fixed buffer. Also, recommend creating the `IPAddress` from the bytes rather than first creating a `string`.